### PR TITLE
feat: add card_eq_NrRealPlaces_add_NrComplexPlaces

### PR DESCRIPTION
**Pull request by riccardobrasca**
We add `card_eq_NrRealPlaces_add_NrComplexPlaces`:  `Fintype.card (InfinitePlace K).card =  NrRealPlaces K + NrComplexPlaces K` for a number field `K`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by xroblot — APPROVED** [review]
LGTM

File: Mathlib/NumberTheory/NumberField/Embeddings.lean [hunk 1]
```diff
@@ -579,6 +584,14 @@ noncomputable abbrev NrRealPlaces := card { w : InfinitePlace K // IsReal w }
 /-- The number of infinite complex places of the number field `K`. -/
 noncomputable abbrev NrComplexPlaces := card { w : InfinitePlace K // IsComplex w }
 
+theorem card_eq_NrRealPlaces_add_NrComplexPlaces : Fintype.card (InfinitePlace K) =
+    NrRealPlaces K + NrComplexPlaces K := by
+  dsimp [InfinitePlace, NrRealPlaces, NrComplexPlaces]
+  convert Fintype.card_subtype_or_disjoint (IsReal (K := K)) (IsComplex (K := K))
+    (disjoint_isReal_isComplex K) using 1
+  symm
+  exact Fintype.card_of_subtype _ (fun w ↦ ⟨fun _ ↦ isReal_or_isComplex w, fun _ ↦ by simp⟩)
```
Comment (MichaelStollBayreuth) [inline Mathlib/NumberTheory/NumberField/Embeddings.lean:593]:
```suggestion
  convert Fintype.card_subtype_or_disjoint (IsReal (K := K)) (IsComplex (K := K))
    (disjoint_isReal_isComplex K) using 1
  exact (Fintype.card_of_subtype _ (fun w ↦ ⟨fun _ ↦ isReal_or_isComplex w, fun _ ↦ by simp⟩)).symm
```
Without the `disimp`, it is actually quite a bit faster...

**Comment by MichaelStollBayreuth** [comment]
maintainer merge

File: Mathlib/NumberTheory/NumberField/Embeddings.lean [hunk 2]
```diff
@@ -579,6 +584,12 @@ noncomputable abbrev NrRealPlaces := card { w : InfinitePlace K // IsReal w }
 /-- The number of infinite complex places of the number field `K`. -/
 noncomputable abbrev NrComplexPlaces := card { w : InfinitePlace K // IsComplex w }
 
+theorem card_eq_NrRealPlaces_add_NrComplexPlaces : Fintype.card (InfinitePlace K) =
```
Comment (eric-wieser) [inline Mathlib/NumberTheory/NumberField/Embeddings.lean:587]:
```suggestion
theorem card_eq_nrRealPlaces_add_nrComplexPlaces : Fintype.card (InfinitePlace K) =
```

File: Mathlib/NumberTheory/NumberField/Embeddings.lean [hunk 3]
```diff
@@ -579,6 +584,12 @@ noncomputable abbrev NrRealPlaces := card { w : InfinitePlace K // IsReal w }
 /-- The number of infinite complex places of the number field `K`. -/
 noncomputable abbrev NrComplexPlaces := card { w : InfinitePlace K // IsComplex w }
```
Comment (eric-wieser) [inline Mathlib/NumberTheory/NumberField/Embeddings.lean:585]: Out of scope, but this should be
```suggestion
noncomputable abbrev nrComplexPlaces : Nat := card { w : InfinitePlace K // IsComplex w }
```

Comment (riccardobrasca) [reply Mathlib/NumberTheory/NumberField/Embeddings.lean:585]: cc @xroblot

Comment (riccardobrasca) [reply Mathlib/NumberTheory/NumberField/Embeddings.lean:585]: I think we can keep it for another PR.

File: Mathlib/NumberTheory/NumberField/Embeddings.lean [hunk 4]
```diff
@@ -414,6 +414,11 @@ theorem isReal_or_isComplex (w : InfinitePlace K) : IsReal w ∨ IsComplex w :=
   rw [← not_isReal_iff_isComplex]; exact em _
 #align number_field.infinite_place.is_real_or_is_complex NumberField.InfinitePlace.isReal_or_isComplex
 
+variable (K) in
+theorem disjoint_isReal_isComplex : Disjoint {(w : InfinitePlace K) | IsReal w}
+    {(w : InfinitePlace K) | IsComplex w} :=
```
Comment (ocfnash) [inline Mathlib/NumberTheory/NumberField/Embeddings.lean:419]: Nitpick:
```suggestion
theorem disjoint_isReal_isComplex :
    Disjoint {(w : InfinitePlace K) | IsReal w} {(w : InfinitePlace K) | IsComplex w} :=
```

Comment (ocfnash) [reply Mathlib/NumberTheory/NumberField/Embeddings.lean:419]: Or even:
```
theorem disjoint_isReal_isComplex :
    Disjoint
      {(w : InfinitePlace K) | IsReal w}
      {(w : InfinitePlace K) | IsComplex w} :=
```

File: Mathlib/NumberTheory/NumberField/Embeddings.lean [hunk 5]
```diff
@@ -579,6 +584,12 @@ noncomputable abbrev NrRealPlaces := card { w : InfinitePlace K // IsReal w }
 /-- The number of infinite complex places of the number field `K`. -/
 noncomputable abbrev NrComplexPlaces := card { w : InfinitePlace K // IsComplex w }
 
+theorem card_eq_nrRealPlaces_add_nrComplexPlaces : Fintype.card (InfinitePlace K) =
+    NrRealPlaces K + NrComplexPlaces K := by
```
Comment (ocfnash) [inline Mathlib/NumberTheory/NumberField/Embeddings.lean:588]:
```suggestion
theorem card_eq_nrRealPlaces_add_nrComplexPlaces : 
    Fintype.card (InfinitePlace K) = NrRealPlaces K + NrComplexPlaces K := by
```

**Review by ocfnash — APPROVED** [review]
Thanks, just optional nitpicks.

bors d+

**Pull request closed without merging**